test/system/staking.test.js

```diff
@@ -5,111 +5,491 @@ const {
   to1e18,
 } = require("../helpers/contract-test-helpers")
 const { initContracts } = require("./init-contracts")
-const { keepManagedGrantAddress } = require("./constants")
-const { helpers } = require("hardhat")
+const {
+  keepGrantStake,
+  keepLiquidTokenStake,
+  keepManagedGrantStake,
+  keepTokenGrantAddress,
+} = require("./constants")
 
 const describeFn =
   process.env.NODE_ENV === "system-test" ? describe : describe.skip
 
-describeFn("System -- staking", () => {
+describeFn("SystemTests: TokenStaking", () => {
   const startingBlock = 13619810
 
-  // Roles.
-  // eslint-disable-next-line no-unused-vars
   let governance
-  let beneficiary
-  let operator
-  let authorizer
   let purse
 
   // Contracts
+  let keepToken
   let keepTokenStaking
+  let keepVendingMachine
   let tokenStaking
+  let mockApplication
 
-  before(async () => {
+  beforeEach(async () => {
     await resetFork(startingBlock)
 
     governance = await ethers.getSigner(0)
```

Comment:
We can but if we want to do that, I'd like to make this change separately from this PR. This will require changes in `init-contracts.js` and all other tests implemented and tests in the PRs that are in the queue (e.g. governance, staking contract upgradeability). This is not a bad idea but it's a bad moment for this change.